td-shim/build.rs

```diff
@@ -43,15 +43,15 @@ fn run_command(mut cmd: Command) {
 }
 
 fn check_environment() -> Result<()> {
-    const CC_ENV_VAR: &str = "CC";
+    const CC_ENV_VAR: &str = "CC_x86_64_unknown_uefi";
     const AS_ENV_VAR: &str = "AS";
-    const AR_ENV_VAR: &str = "AR";
+    const AR_ENV_VAR: &str = "AR_x86_64_unknown_uefi";
 
     // Defaults to 'cc' but also honours the CC_ENV_VAR environment variable.
     let cfg = cc::Build::new().try_get_compiler()?;
 
     // GCC cannot be used as it fails to link.
-    let required_compiler = "clang";
+    let required_compiler = "cl";
```

Comment:
`clang` is  required to build ring when `--target` is `x86_64-unknown-uefi`).  I don't think you should change this.